src/NServiceBus.Core/Transports/Learning/DirectoryBasedTransaction.cs

```diff
@@ -66,17 +64,67 @@ public Task Enlist(string messagePath, string messageContents)
         public void Complete()
         {
             if (!committed)
-            {
                 return;
-            }
             foreach (var outgoingFile in outgoingFiles)
-            {
                 File.Move(outgoingFile.TxPath, outgoingFile.TargetPath);
-            }
 
             Directory.Delete(commitDir, true);
         }
 
+        public static void RecoverPartiallyCompletedTransactions(string basePath)
+        {
+            var pendingRootDir = Path.Combine(basePath, PendingDirName);
+
+            if (Directory.Exists(pendingRootDir))
+            {
+                foreach (var transactionDir in new DirectoryInfo(pendingRootDir).EnumerateDirectories())
+                {
+                    var transaction = new DirectoryBasedTransaction(basePath, transactionDir.Name, false);
```

Comment:
it feels weird tbh. As this seems to be a more subjective opinion, I wouldn't consider this a blocker but I think I'd prefer it to not create a transaction in order to recover pending transactions.